amlb/benchmark.py

```diff
@@ -360,32 +361,45 @@ def __json__(self):
         self.cores = min(self.cores, sys_cores) if self.cores > 0 else sys_cores
         log.info("Assigning %s cores (total=%s) for new task %s.", self.cores, sys_cores, self.name)
 
         sys_mem = system_memory_mb()
-        os_recommended_mem = rconfig().benchmarks.os_mem_size_mb
+        os_recommended_mem = ns.get(rconfig(), f"{mode}.os_mem_size_mb", rconfig().benchmarks.os_mem_size_mb)
         left_for_app_mem = int(sys_mem.available - os_recommended_mem)
         assigned_mem = round(self.max_mem_size_mb if self.max_mem_size_mb > 0
                              else left_for_app_mem if left_for_app_mem > 0
                              else sys_mem.available)
         log.info("Assigning %.f MB (total=%.f MB) for new %s task.", assigned_mem, sys_mem.total, self.name)
         self.max_mem_size_mb = assigned_mem
-        if assigned_mem > sys_mem.available:
-            log.warning("WARNING: Assigned memory (%(assigned).f MB) exceeds system available memory (%(available).f MB / total=%(total).f MB)!",
-                        dict(assigned=assigned_mem, available=sys_mem.available, total=sys_mem.total))
+        if assigned_mem > sys_mem.total:
+            handle_unfulfilled(f"Total system memory {sys_mem.total} MB doesn't meet requirements ({assigned_mem} MB)!.",
+                               on_auto='fail')
+        elif assigned_mem > sys_mem.available:
+            handle_unfulfilled(f"Assigned memory ({assigned_mem} MB) exceeds system available memory ({sys_mem.available} MB / total={sys_mem.total} MB)!")
         elif assigned_mem > sys_mem.total - os_recommended_mem:
-            log.warning("WARNING: Assigned memory (%(assigned).f MB) is within %(buffer).f MB of system total memory (%(total).f MB): "
-                        "We recommend a %(buffer).f MB buffer, otherwise OS memory usage might interfere with the benchmark task.",
-                        dict(assigned=assigned_mem, available=sys_mem.available, total=sys_mem.total, buffer=os_recommended_mem))
+            handle_unfulfilled(f"Assigned memory ({assigned_mem} MB) is within {sys_mem.available} MB of system total memory {sys_mem.total} MB): "
+                               f"We recommend a {os_recommended_mem} MB buffer, otherwise OS memory usage might interfere with the benchmark task.")
 
         if self.min_vol_size_mb > 0:
             sys_vol = system_volume_mb()
             os_recommended_vol = rconfig().benchmarks.os_vol_size_mb
             if self.min_vol_size_mb > sys_vol.free:
-                log.warning("WARNING: Available volume memory (%(available).f MB / total=%(total).f MB) doesn't meet requirements (%(required).f MB)!",
-                            dict(required=self.min_vol_size_mb+os_recommended_vol, available=sys_vol.free, total=sys_vol.total))
+                handle_unfulfilled(f"Available volume memory ({sys_vol.free} MB / total={sys_vol.total} MB) doesn't meet requirements ({self.min_vol_size_mb+os_recommended_vol} MB)!")
```

Comment:
Should we use something akin to 'disk space' or 'storage' to further avoid confusion?

Reply:
you mean renaming `min_vol_size_mb` to `min_disk_space_mb`?
I can do that if you think it's better: I recently introduced some logic allowing to rename config entries in a backwards compatible way, so I don't mind renaming.
I also have the idea to remove all those unit suffixes (for time, bytes…) and write:
```
min_disk_space = 64gb
```
instead of
```
min_disk_space_mb = 65536
```
Will probably do this later as it's not urgent, and I want to do it for all units, so maybe it's worth waiting for that task and switch to the new name at the same time.

Reply:
> Should we use something akin to 'disk space' or 'storage' to further avoid confusion?

Oh, just realized that you were probably talking about the message: "Available disk space", correct? Yes, it would be less confusing indeed.

Reply:
Yes, sorry for being too brief there. The field name was clear, I foresaw misinterpretations of the warning message.

Reply:
I think declaring `disk_space = 64gb` in a configuration is nicer (because it allows the user to specify unit size). But I actually kind of like units in variable names if it's otherwise not immediately clear what unit is used. But as long as it's easy enough to deduce what unit is used without running the code I'm ok with anything.